staging/kos/pkg/controllers/connectionagent/connection_agent.go

```diff
@@ -0,0 +1,1180 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package connectionagent
+
+import (
+	"fmt"
+	gonet "net"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	k8scorev1api "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfields "k8s.io/apimachinery/pkg/fields"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	k8sutilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	k8swait "k8s.io/apimachinery/pkg/util/wait"
+	k8scorev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	k8scache "k8s.io/client-go/tools/cache"
+	k8seventrecord "k8s.io/client-go/tools/record"
+	k8sworkqueue "k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	netv1a1 "k8s.io/examples/staging/kos/pkg/apis/network/v1alpha1"
+	kosclientset "k8s.io/examples/staging/kos/pkg/client/clientset/versioned"
+	kosscheme "k8s.io/examples/staging/kos/pkg/client/clientset/versioned/scheme"
+	netvifc1a1 "k8s.io/examples/staging/kos/pkg/client/clientset/versioned/typed/network/v1alpha1"
+	kosinformers "k8s.io/examples/staging/kos/pkg/client/informers/externalversions"
+	koslisterv1a1 "k8s.io/examples/staging/kos/pkg/client/listers/network/v1alpha1"
+	netfabric "k8s.io/examples/staging/kos/pkg/networkfabric"
+	"k8s.io/examples/staging/kos/pkg/util/parse"
+)
+
+const (
+	// localAttsCacheID is the ID for the local NetworkAttachments informer's
```

Comment:
e84a065 eliminates the cacheID type, but in the variable names listerID is used instead of VNI (no new listerID type is present though, `uint32` are used).